### PR TITLE
fix: revert 4410 and fix silent error

### PR DESCRIPTION
**Pull request by hanxiao**
Goals:

- ...
- ...
- [ ] check and update documentation. See [guide](https://github.com/jina-ai/jina/CONTRIBUTING.md#documentation-guidelines) and ask the team.

### CODE REVIEW

File: jina/clients/base/http.py [hunk 1]
```diff
@@ -93,80 +93,59 @@ async def _get_results(
...
-                        on_done=on_done,
-                        on_always=on_always,
-                        continue_on_error=self.continue_on_error,
-                        logger=self.logger,
-                    )
-                    if self.show_progress:
-                        p_bar.update()
-                    yield resp
-
-            except (aiohttp.ClientError, ValueError, ConnectionError) as e:
```
Comment (JoanFM) [inline jina/clients/base/http.py:154]: Please do not change this, the `Exception` should be properly handled. @JohannesMessner

Comment (hanxiao) [reply jina/clients/base/http.py:154]: no, they should bn raised

File: jina/resources/extra-requirements.txt [hunk 2]
```diff
@@ -46,6 +46,7 @@ filelock:                   standard,devel
 requests:                   standard,devel
 websockets:                 standard,devel
 pydantic:                   standard,devel
+importlib_metadata:         core
```
Comment (JoanFM) [inline jina/resources/extra-requirements.txt:49]: why is this needed?

File: extra-requirements.txt [hunk 3]
```diff
@@ -46,6 +46,7 @@ filelock:                   standard,devel
 requests:                   standard,devel
 websockets:                 standard,devel
 pydantic:                   standard,devel
+importlib_metadata:         core
```
Comment (JoanFM) [inline extra-requirements.txt:49]:
```suggestion
```

**Comment by hanxiao** [comment]
This is insane: making those flaky tests work takes more time than the PR itself. Those flaky tests should be completely deleted.

**Comment by JohannesMessner** [comment]
> This is insane: making those flaky tests work takes more time than the PR itself. Those flaky tests should be completely deleted.

The flakyness only really seems to happen on this PR, and I don't know why. I don't see the connection. But usually this is not a big problem.

**Comment by hanxiao** [comment]
I will wait the test for the last time, if it is still timeout and we have no clear clue then I will merge as is
